modules/invibesBidAdapter.js

```diff
@@ -34,18 +34,26 @@ export const spec = {
   interpretResponse: function (responseObj, requestParams) {
     return handleResponse(responseObj, requestParams != null ? requestParams.bidRequests : null);
   },
-  getUserSyncs: function(syncOptions) {
+  getUserSyncs: function (syncOptions) {
     if (syncOptions.iframeEnabled) {
+      handlePostMessage();
+      const syncUrl = buildSyncUrl();
+
       return {
         type: 'iframe',
-        url: CONSTANTS.SYNC_ENDPOINT
+        url: syncUrl
       };
     }
   }
 };
 
 registerBidder(spec);
 
+// some state info is required: cookie info, unique user visit id
+const topWin = getTopMostWindow();
+let invibes = topWin.invibes = topWin.invibes || {};
```

Comment:
What's the reason for creating the invibes object on the top window and attaching some of the functions below to it?

Reply:
The location is on the top window so that we don't initialize our local id on this page twice. The cookie info and the id are passed to the user sync function. Also the functions for cookies are reused inside the postMessage, so it made sense to just put them all inside the same object